jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/Kubernetes/KubernetesAddOnEnabledRule.java

```diff
@@ -0,0 +1,119 @@
+package com.tmobile.cloud.azurerules.Kubernetes;
+
+import com.amazonaws.util.StringUtils;
+import com.google.common.collect.HashMultimap;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.tmobile.cloud.awsrules.utils.PacmanUtils;
+import com.tmobile.cloud.awsrules.utils.RulesElasticSearchRepositoryUtil;
+import com.tmobile.cloud.constants.PacmanRuleConstants;
+import com.tmobile.pacman.commons.PacmanSdkConstants;
+import com.tmobile.pacman.commons.exception.InvalidInputException;
+import com.tmobile.pacman.commons.exception.RuleExecutionFailedExeption;
+import com.tmobile.pacman.commons.rule.Annotation;
+import com.tmobile.pacman.commons.rule.BaseRule;
+import com.tmobile.pacman.commons.rule.PacmanRule;
+import com.tmobile.pacman.commons.rule.RuleResult;
+import com.tmobile.pacman.commons.utils.CommonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+@PacmanRule(key = "enable-addon-policies-for-kubernetes", desc = "Ensure AKS uses Azure policies add-on", severity = PacmanSdkConstants.SEV_LOW, category = PacmanSdkConstants.SECURITY)
+public class KubernetesAddOnEnabledRule extends BaseRule {
+
+    private static final Logger logger = LoggerFactory.getLogger(KubernetesAddOnEnabledRule.class);
+    @Override
+    public RuleResult execute(Map<String, String> ruleParam, Map<String, String> resourceAttributes) {
+        logger.info("Executing KubernetesAddOnEnabledRule..");
+        String severity = ruleParam.get(PacmanRuleConstants.SEVERITY);
+        String category = ruleParam.get(PacmanRuleConstants.CATEGORY);
+        String successMSG = ruleParam.get(PacmanRuleConstants.SUCCESS);
+        String failureMsg = ruleParam.get(PacmanRuleConstants.VIOLATION_REASON);
+        if (!PacmanUtils.doesAllHaveValue(severity, category)) {
+            logger.info(PacmanRuleConstants.MISSING_CONFIGURATION);
+            throw new InvalidInputException(PacmanRuleConstants.MISSING_CONFIGURATION);
+        }
+        String esUrl = CommonUtils.getEnvVariableValue(PacmanSdkConstants.ES_URI_ENV_VAR_NAME);
+        String url = CommonUtils.getEnvVariableValue(PacmanSdkConstants.ES_URI_ENV_VAR_NAME);
+        if (!StringUtils.isNullOrEmpty(url)) {
+            esUrl = url + "/azure_kubernetes/_search";
+        }
+        boolean isValid = false;
+        Map<String, Object> mustFilter = new HashMap<>();
+        mustFilter.put(PacmanRuleConstants.LATEST, true);
```

Comment:
To retrieve only the resource data shouldn't we add resourceid in the filer,as done for other policies.

Reply:
Done